drivers/include/pulse_counter.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_pulse_counter sensor
+ * @ingroup     drivers_sensors
+ *
+ * The connection between the MCU and the PULSE_COUNTER is based on the
+ * GPIO-interface.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Driver for the PULSE_COUNTER sensor
+ *
+ * @author      Hyung-Sin <hs.kim@cs.berkeley.edu>
+ */
+
+#ifndef PULSE_COUNTER_H_
+#define PULSE_COUNTER_H_
+
+#include <stdint.h>
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    gpio_t    gpio;            /**< GPIO pin that sensor is connected to */
+} pulse_counter_params_t;
+
+/**
+  * @brief   Device descriptor for a PULSE_COUNTER device
+  * @{
+  */
+typedef struct {
+    uint8_t   id;             /**< pulse counter id to distinguish multiple counters */
+    pulse_counter_params_t p;
+} pulse_counter_t;
+/** @} */
+
+/**
+ * @brief   Initialize an PULSE_COUNTER device
+ *
+ * @param[out] dev          device descriptor
+ * @param[in] params        GPIO bus the device is connected to
+ *
+ * The valid address range is 0x1E - 0x1F depending on the configuration of the
+ * address pins SA0 and SA1.
+ *
+ * @return                   0 on success
+ * @return                  -1 on error
+ * @return                  -2 on invalid address
```

Comment:
The implemented function currently never returns `-2`.

Either the documentation, or the implementation needs to be updated.

Reply:
fixed. Thanks!